# Verificación del preprocesamiento vigente

In [1]:
from pathlib import Path
import runpy
import pandas as pd
from IPython.display import display
ROOT = next(p for p in [Path.cwd(), *Path.cwd().parents] if (p/'src/17_validate_pipeline.py').is_file())
module=runpy.run_path(str(ROOT/'src/17_validate_pipeline.py'))
RECALCULAR=False
if RECALCULAR:
    module['main']()
module['render'](False)
display(pd.read_csv(ROOT/'outputs/tables/preprocessing_full_audit.csv'))


,symbol,fold,features,n_train,n_validation,max_abs_scaled_train_mean,mean_lag0,scale_lag0,validation_does_not_change_scaler,inverse_recovers_validation
0,BNBUSDT,1,168,5781,6370,5.427859e-15,156.059699,189.072330,True,True
1,BTCUSDT,1,168,5781,6370,4.548938e-15,30648.452659,18309.962940,True,True
2,ETHUSDT,1,168,5781,6370,4.395339e-15,1244.166873,980.352038,True,True
3,SOLUSDT,1,168,5781,6370,3.391740e-15,11.062052,13.362181,True,True
4,XRPUSDT,1,168,5781,6370,4.488481e-15,0.527859,0.403743,True,True
5,BNBUSDT,2,168,12342,6951,5.112666e-15,302.018696,201.081170,True,True
6,BTCUSDT,2,168,12342,6951,8.939022e-15,38390.371042,15972.513829,True,True
7,ETHUSDT,2,168,12342,6951,9.323238e-15,2272.509131,1304.819575,True,True
8,SOLUSDT,2,168,12342,6951,4.789557e-15,67.736325,71.910541,True,True
9,XRPUSDT,2,168,12342,6951,8.376773e-15,0.712706,0.356901,True,True


### 2.9.7 Verificación ejecutada

Se comprueban los 25 bloques de entrenamiento de la sección 3, con 168 entradas y confinamiento de historia y objetivo. Para cada uno se verifica que las medias del escalador coincidan con TRAIN, que transformar validación no modifique medias, varianzas ni escalas, que las salidas sean finitas y que la transformación inversa recupere los cierres. Una perturbación artificial de validación confirma que se aplican los mismos parámetros ya aprendidos. No se exige media cero en validación.

| Activo | Fold | n TRAIN | n VALIDATION | Máxima media TRAIN estandarizada absoluta |
| --- | --- | --- | --- | --- |
| BNBUSDT | 1 | 5781 | 6370 | 5.43e-15 |
| BTCUSDT | 1 | 5781 | 6370 | 4.55e-15 |
| ETHUSDT | 1 | 5781 | 6370 | 4.40e-15 |
| SOLUSDT | 1 | 5781 | 6370 | 3.39e-15 |
| XRPUSDT | 1 | 5781 | 6370 | 4.49e-15 |
| BNBUSDT | 2 | 12342 | 6951 | 5.11e-15 |
| BTCUSDT | 2 | 12342 | 6951 | 8.94e-15 |
| ETHUSDT | 2 | 12342 | 6951 | 9.32e-15 |
| SOLUSDT | 2 | 12342 | 6951 | 4.79e-15 |
| XRPUSDT | 2 | 12342 | 6951 | 8.38e-15 |
| BNBUSDT | 3 | 19484 | 6758 | 4.90e-15 |
| BTCUSDT | 3 | 19484 | 6758 | 4.33e-15 |
| ETHUSDT | 3 | 19484 | 6758 | 9.15e-15 |
| SOLUSDT | 3 | 19484 | 6758 | 5.56e-15 |
| XRPUSDT | 3 | 19484 | 6758 | 1.30e-14 |
| BNBUSDT | 4 | 26433 | 6951 | 5.33e-15 |
| BTCUSDT | 4 | 26433 | 6951 | 1.32e-14 |
| ETHUSDT | 4 | 26433 | 6951 | 1.68e-14 |
| SOLUSDT | 4 | 26433 | 6951 | 9.55e-15 |
| XRPUSDT | 4 | 26433 | 6951 | 1.45e-14 |
| BNBUSDT | 5 | 33575 | 6951 | 2.16e-14 |
| BTCUSDT | 5 | 33575 | 6951 | 7.10e-15 |
| ETHUSDT | 5 | 33575 | 6951 | 2.05e-14 |
| SOLUSDT | 5 | 33575 | 6951 | 2.31e-14 |
| XRPUSDT | 5 | 33575 | 6951 | 2.87e-14 |

Los 25 controles se superan. Esto verifica propiedades concretas del preprocesamiento, no una garantía universal de ausencia de fuga. [Auditoría completa](../outputs/tables/preprocessing_full_audit.csv).

### 2.9.8 Reproducibilidad

Ejecutar `python src/17_validate_pipeline.py`. El script comparte preparación, fronteras y clase de escalado con el modelo; no utiliza TEST. Los metadatos conservan las huellas de DEVELOPMENT y de los folds. La comprobación histórica de una única columna en `src/17_preprocessing_close.py` queda sustituida por esta verificación del protocolo vigente. [Notebook ejecutado](./17_preprocessing_close.ipynb).
